docker/ci/test_migrations.sh

```diff
@@ -3,139 +3,212 @@
+# Find all the info we need about the first migration in the git history.
+gh_group "Finding first migration"
+run_in_container "webapp" alembic history -r'base:base+1' -v
+# Debug echo is disabled since we are capturing the output of the command
+DEBUG_ECHO_ENABLED=0
+first_migration_id=$(run_in_container "webapp" alembic history -r'base:base+1' -v | head -n 1 | cut -d ' ' -f2)
+DEBUG_ECHO_ENABLED=1
 if [[ -z $first_migration_id ]]; then
-  echo "ERROR: Could not find first migration id."
-  exit 1
+  error_and_cleanup "Could not find first migration id." 1
 fi
 
 first_migration_file=$(find alembic/versions -name "*${first_migration_id}*.py")
 if [[ -z $first_migration_file ]]; then
-  echo "ERROR: Could not find first migration file."
-  exit 1
+  error_and_cleanup "Could not find first migration file." 1
 fi
 
-echo "First migration file: ${first_migration_file}"
-echo ""
-
-# Find the git commit where the first migration file was added
 first_migration_commit=$(git log --follow --format=%H --reverse "${first_migration_file}" | head -n 1)
 if [[ -z $first_migration_commit ]]; then
-  echo "ERROR: Could not find first migration commit hash."
-  exit 1
+  error_and_cleanup "Could not find first migration commit hash." 1
+fi
+first_migration_container="ghcr.io/thepalaceproject/circ-webapp:sha-${first_migration_commit:0:7}"
+echo "First migration info:"
+echo "  id: ${first_migration_id}"
+echo "  file: ${first_migration_file}"
+echo "  commit: ${first_migration_commit}"
+echo "  container: ${first_migration_container}"
+
+container_image=$(sed -n 's/^ *image: "\(.*\)"/\1/p' docker/ci/test_migrations.yml)
+if [[ -z $container_image ]]; then
+  error_and_cleanup "Could not find container image in test_migrations.yml" 1
+fi
+
+if [[ "$container_image" != "$first_migration_container" ]]; then
+  error_and_cleanup "Incorrect container image in test_migrations.yml. Please update." 1
 fi
+gh_endgroup
 
-echo "Starting containers"
+gh_group "Starting service containers"
 compose_cmd down
-compose_cmd up -d pg
+compose_cmd up -d pg os
+gh_endgroup
 
-echo "Initializing database at commit ${first_migration_commit}"
-git checkout -q "${first_migration_commit}"
-run_in_container "./bin/util/initialize_instance"
+gh_group "Initializing database"
+run_in_container "webapp-old" "./bin/util/initialize_instance"
 initialize_exit_code=$?
 if [[ $initialize_exit_code -ne 0 ]]; then
-  echo "ERROR: Failed to initialize instance."
-  cleanup
-  exit $initialize_exit_code
+  error_and_cleanup "Failed to initialize instance." $initialize_exit_code
 fi
-echo ""
+gh_endgroup
 
 # Migrate up to the current commit and check if the database is in sync
-echo "Testing upgrade migrations on branch ${current_branch}"
-git checkout -q "${current_branch}"
-run_migrations "upgrade head"
-check_db "A new migration is required or a up migration is broken."
+gh_group "Testing upgrade migrations"
+run_migrations "webapp" "upgrade head"
```

Comment:
Should this be" webapp-old" here?

Reply:
It can't be `webapp-old` because the old container doesn't have the migrations in it in order to move to the latest DB version.

Reply:
okay - right of course.